IDR/notebook/scripts/nbconverted/1.describe_screens.py

```diff
@@ -140,15 +165,65 @@ def describe_screen(screen_id):
                 # Clean channels value and separate into stain and stain target
                 stain = set()
                 stain_target = set()
-                for entry in channels.split(";"):
-                    temp_list = entry.split(":")
-                    stain.add(temp_list[0])
-                    stain_target.add(temp_list[1])
+                for channel in channels.split(";"):
+                    # For channel entries with 'stain:target' format
+                    if bool(re.search(r"([\:])+", channel)):
+                        # Separate stain and target
+                        st_list = channel.split(":")
+                        # Add to respective sets
+                        stain.add(st_list[0])
```

Comment:
Why are you creating a set? Are sets ordered? I'm not sure how you're using this functionality (I will continue reading) but if you're appending later using a set instead of a list could cause misalignment errors.

Reply:
@gwaybio I had an incorrect thought process behind creating a set. I'm realizing now that ordered lists would be more accurate representations of the metadata. I switched the sets to lists.

Added in f99ffc6cd21dabce7e89f668ef858b74cc87e44b.

Reply:
We discussed on the call 1) the merits of using a set here and 2) instead of separating `stain` and `target` to standardize `stain:target` as a single column.

Parker is going to think through the implications and decide how to proceed.

Reply:
@gwaybio After thinking some more about this, I decided to combine stain and target into a single column and analyze diversity at the perturbation level. The new `stain_target` will consist of a string of alphabetically sorted `stain:target` combinations split by a semi-colon. Ex: `'stain:target ; stain:target ; stain:target'`. Alphabetizing in a list and then combining elements into a string will prevent false unique entries like the following: `GFP:Cytoplasm ; Cherry:Nucleus` vs `Cherry:Nucleus ; GFP:Cytoplasm` where they're the same combination of channels, but different permutations. The standardized entries would be `Cherry:Nucleus ; GFP:Cytoplasm` and both would be considered as one unique category, not two.